src/core/skills/characters/yijiang2014/shibei.ts

```diff
@@ -0,0 +1,56 @@
+import { GameEventIdentifiers, ServerEventFinder } from 'core/event/event';
+import { AllStage, DamageEffectStage, PhaseChangeStage, PlayerPhase } from 'core/game/stage_processor';
+import { Player } from 'core/player/player';
+import { Room } from 'core/room/room';
+import { CompulsorySkill, ShadowSkill, TriggerSkill } from 'core/skills/skill';
+
+@CompulsorySkill({ name: 'shibei', description: 'shibei_description' })
+export class ShiBei extends TriggerSkill {
+  isTriggerable(event: ServerEventFinder<GameEventIdentifiers.DamageEvent>, stage?: AllStage) {
+    return stage === DamageEffectStage.AfterDamageEffect;
+  }
+
+  public canUse(room: Room, owner: Player, event: ServerEventFinder<GameEventIdentifiers.DamageEvent>): boolean {
+    return event.toId === owner.Id;
+  }
+
+  public async onTrigger(): Promise<boolean> {
+    return true;
+  }
+
+  public async onEffect(room: Room, event: ServerEventFinder<GameEventIdentifiers.SkillEffectEvent>): Promise<boolean> {
+    if (room.getFlag<boolean>(event.fromId, this.Name) !== true) {
```

Comment:
no need to sync up flags on client side, since it's only working on server side 
```suggestion
    if (room.getPlayerById(event.fromId).getFlag<boolean>(this.Name) !== true) {
```